src/main/java/org/mskcc/cbio/oncokb/web/rest/CompanyResource.java

```diff
@@ -174,4 +191,39 @@ public ResponseEntity<Boolean> verifyCompanyName(@Valid @RequestBody VerifyCompa
         }
         return new ResponseEntity<>(isValid, HttpStatus.OK);
     }
+
+    /**
+     * {@code DELETE  /companies/:id} : delete the "id" company.
+     *
+     * @param id the id of the companyDTO to delete.
+     * @return the {@link ResponseEntity} with status {@code 204 (NO_CONTENT)}.
+     */
+    @DeleteMapping("/companies/{id}")
+    public ResponseEntity<Void> deleteCompany(@PathVariable Long id) {
+        log.debug("REST request to delete Company : {}", id);
+        Optional<CompanyDTO> maybeCompanyDTO = companyService.findOne(id);
+        if (!maybeCompanyDTO.isPresent()) {
+            return ResponseEntity.notFound().build();
+        }
+
+        CompanyDTO companyDTO = maybeCompanyDTO.get();
+        Boolean isCommercial = companyDTO.getLicenseType().equals(LicenseType.COMMERCIAL);
+        List<UserDetails> userDetails = userDetailsRepository.findByCompanyId(id);
+        for (UserDetails userDetail : userDetails) {
+            userDetail.setCompany(null);
+            userDetail.setCompanyName(null);
+        }
+
+        if (isCommercial) {
+            List<User> users = userDetails.stream().map(UserDetails::getUser).collect(Collectors.toList());
+            for (User user : users) {
+                user.setActivated(false);
+            }
+            userRepository.saveAll(users);
+        }
+
+        userDetailsRepository.saveAll(userDetails);
+        companyService.delete(id);
```

Comment:
These should be moved to company service and make it transactional. https://www.baeldung.com/transaction-configuration-with-jpa-and-spring